apps/challenges/serializers.py

```diff
@@ -102,11 +106,22 @@ class ZipChallengeSerializer(ChallengeSerializer):
     """
     Serializer used for creating challenge through zip file.
     """
+    def __init__(self, *args, **kwargs):
+        super(ZipChallengeSerializer, self).__init__(*args, **kwargs)
+
+        context = kwargs.get('context')
+        if context:
+            image = context.get('image')
+            evaluation_script = context.get('evaluation_script')
+
```

Comment:
No, this isn't new feature. it is to offload all the saving task to the serializers. Please refer [here](https://github.com/Cloud-CV/EvalAI/pull/1037#discussion-diff-123895061R489)